### PR TITLE
Support the receive function as an additional operation type

### PR DESCRIPTION
**Pull request by fordN**
In the Solidity 0.6.0 release the unnamed function, previously `fallback`, has been split into two functions, `receive` and `fallback`.   ([0.6.0 breaking changes](https://solidity.readthedocs.io/en/v0.6.0/060-breaking-changes.html))

This PR adds support for the `receive` function.

### CODE REVIEW

File: ethabi/src/contract.rs [hunk 1]
```diff
@@ -47,12 +49,14 @@ impl<'a> Visitor<'a> for ContractVisitor {
...
-		A: SeqAccess<'a>,
-	{
-		let mut result =
-			Contract { constructor: None, functions: HashMap::default(), events: HashMap::default(), fallback: false };
+	fn visit_seq<A>(self, mut seq: A) -> Result<Self::Value, A::Error> where A: SeqAccess<'a> {
+		let mut result = Contract {
+			constructor: None,
+			functions: HashMap::default(),
+			events: HashMap::default(),
+			receive: true,
```
Comment (gakonst) [inline ethabi/src/contract.rs:57]: Should the default `receive` value be true?

Comment (dvdplm) [reply ethabi/src/contract.rs:57]: I have the same question. Reading [this](https://solidity.readthedocs.io/en/v0.6.0/060-breaking-changes.html#semantic-and-syntactic-changes) I'd say it should be `false`.
Somewhat related q: why not impl `Default` for `Contract`?

File: ethabi/src/contract.rs [hunk 2]
```diff
@@ -116,11 +123,6 @@ impl Contract {
 	pub fn events(&self) -> Events {
 		Events(self.events.values().flatten())
 	}
-
-	/// Returns true if contract has fallback
-	pub fn fallback(&self) -> bool {
```
Comment (dvdplm) [inline ethabi/src/contract.rs:121]: Why remove this?

Comment (fordN) [reply ethabi/src/contract.rs:121]: It's not used anywhere.

File: ethabi/src/contract.rs (hunk 1 above)
Comment (fordN) [reply ethabi/src/contract.rs:57]: Hmm, good point, I'm honestly not sure why I thought `true` was the right default here.  It should indeed be `false`. 

As for impl `Default`, with this PR I was trying to be as unintrusive as possible to get it in quickly, but we are way past that point ;).  Looking at it now: the default values for `Contract` properties are consistent with the defaults for each property type so `Default` can be derived.

**Comment by gakonst** [comment]
@dvdplm could you take another look at this? Changes seems straightforward to me

File: ethabi/src/operation.rs [hunk 3]
```diff
@@ -25,6 +25,7 @@ pub enum Operation {
 	Event(Event),
 	/// Fallback, ignored.
 	Fallback,
+	Receive,
```
Comment (dvdplm) [inline ethabi/src/operation.rs:28]: Please document.

**Comment by tomusdrw** [comment]
@fordN could you please add a test case and docs @dvdplm requested?

**Comment by smessmer** [comment]
is there any progress on this? This is blocking https://github.com/tomusdrw/rust-web3/issues/410 which in turn blocks one of my projects.

**Comment by vkgnosis** [comment]
Merged in 90659eb0b260ea079ca7a5e52a8ca04c2e2a810d

**Pull request closed without merging**